be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/AnswerGenerator.scala

```diff
@@ -19,18 +19,22 @@ object AnswerGenerator extends AskPatternConstants {
 
     case Left(rpcRequest: JsonRpcRequest) => rpcRequest.getParams match {
       case Catchup(channel) =>
-        val f: Future[GraphMessage] = (DbActor.getInstance ? DbActor.Catchup(channel)).map {
+        val ask: Future[GraphMessage] = (DbActor.getInstance ? DbActor.Catchup(channel)).map {
           case DbActor.DbActorCatchupAck(list: List[Message]) =>
             val resultObject: ResultObject = new ResultObject(list)
             Left(JsonRpcResponse(RpcValidator.JSON_RPC_VERSION, Some(resultObject), None, rpcRequest.id))
           case DbActor.DbActorNAck(code, description) => Right(PipelineError(code, description, rpcRequest.id))
           case _ => Right(PipelineError(ErrorCodes.SERVER_ERROR.id, "Database actor returned an unknown answer", rpcRequest.id))
         }
 
-        Await.result(f, duration)
+        Await.result(ask, duration)
 
 
-      case Broadcast(_, _) => ???
+      case Broadcast(_, _) => Right(PipelineError(
+        ErrorCodes.SERVER_ERROR.id,
+        "Server received a Broadcast message which should never happen (broadcast messages are only emitted by server)",
```

Comment:
This is not going to remain true when server-to-server communication gets implemented.
A comment might be warranted :)